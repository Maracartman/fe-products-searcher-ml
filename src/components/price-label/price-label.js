import React from 'react';
import './price-label.scss';
import cx from 'classnames';
import PropTypes from 'prop-types';

export const PriceLabel = ({
  children,
  keyId,
  float,
  showFloating,
  increaseSize
}) => (
  <div className={'presentational-price__container'}>
    <p
      key={keyId}
      role={'presentation'}
      className={cx({
        price__label: !increaseSize,
        'price__label--increased': increaseSize
      })}
    >
      {children}
    </p>
    {showFloating && (
      <p
        key={`${keyId}--float`}
        role={'presentation'}
        className={cx({
          'price__label--digits': !increaseSize,
          'price__label--digits--increased': increaseSize
        })}
      >
        {float}
      </p>
    )}
  </div>
);

PriceLabel.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.number,
    PropTypes.string
  ]).isRequired,
  keyId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  float: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  showFloating: PropTypes.bool,
  increaseSize: PropTypes.bool
};

PriceLabel.defaultProps = {
  increaseSize: false,
  showFloating: false,
  float: '',
  keyId: Math.random() / 12 + 1333 * Math.random()
};
