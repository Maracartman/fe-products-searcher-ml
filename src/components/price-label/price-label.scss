.presentational-price__container {
    display: inline-flex;
    p {
        margin: 0;
    }
}



p.price__label {
    font-size: 24px;

    &--increased {
        font-size: 46px;
    }

    &--digits {
        font-size: 10px;
        margin-top: 7px;
    }

    &--digits--increased {
        font-size: 15px;
        margin-top: 14px;
    }
}