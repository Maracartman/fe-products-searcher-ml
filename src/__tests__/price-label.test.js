import React from 'react';
import { PriceLabel } from '../components/price-label/price-label';
import { shallow } from 'enzyme';

describe('PriceLabel test suite', () => {
  const priceSample = '15.000';
  const float = '99';
  it('Should loading match Snapshot', () => {
    const price = shallow(<PriceLabel float={float}>{priceSample}</PriceLabel>);
    expect(price).toMatchSnapshot();
  });
  it('Should PriceLabel render expected price', () => {
    const price = shallow(<PriceLabel float={float}>{priceSample}</PriceLabel>);
    expect(price.contains(priceSample)).toEqual(true);
  });
  it('Should PriceLabel not render floats', () => {
    const price = shallow(<PriceLabel float={float}>{priceSample}</PriceLabel>);
    expect(price.contains(float)).toEqual(false);
  });

  it('Should PriceLabel render floats', () => {
    const price = shallow(
      <PriceLabel float={float} showFloating>
        {priceSample}
      </PriceLabel>
    );
    expect(price.contains(float)).toEqual(true);
  });

  it('Should PriceLabel show increased price amount', () => {
    const price = shallow(
      <PriceLabel float={float} increaseSize>
        {priceSample}
      </PriceLabel>
    );
    console.log(price.html())
    expect(price.find('.price__label--increased').length)
    .toBeGreaterThan(0);
  });
});
