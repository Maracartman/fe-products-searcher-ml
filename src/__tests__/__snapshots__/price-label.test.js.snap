// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceLabel test suite Should loading match Snapshot 1`] = `ShallowWrapper {}`;
